### ISSUE TITLE
Unbounded actors created for log when console is stuck

### ISSUE THREAD

**Issue by neeraj9** [erlang-lager/lager#466]
While using lager 06f676aad8ed823b7838a70288cd4d13b5a25ede one of the production cluster node started consuming a lot of memory. Upon detailed inspection, which is covered as part of my bug report https://bugs.erlang.org/browse/ERL-710 it was found that lager (https://github.com/erlang-lager/lager/blob/bf60290101306b3cc26544d0d181a5cff8931e82/include/lager.hrl#L137) spawns unbounded actors. This is an issue when gen_event:which_handlers(lager_event) is stuck, which happened in this particular case.

Although the real issue is still under investigation (as part of https://bugs.erlang.org/browse/ERL-710), but it is better to protect against such issues by keeping spawn under control and bounded. There should be a mechanism to break the circuit when io write is stuck instead of allowing unbounded actors to be spawned for each log.

**Comment by Vagabond** [comment]
Yeah, that's fair. Usually I'd recommend disabling console logging in production because the console is generally a problematic device to work with because of exactly this issue.

**Comment by mrallen1** [comment]
It's a fair point that lager ought to use a proxy process rather than a blind spawn in this situation.

**Comment by dingdongnigetou** [comment]
any improvment ? we aslo get thie issue :(
